### PR TITLE
VMware: add new module: vmware_host_acceptance

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

lib/ansible/modules/cloud/vmware/vmware_host_acceptance.py
test/integration/targets/vmware_host_acceptance/aliases
test/integration/targets/vmware_host_acceptance/tasks/main.yml

##### ANSIBLE VERSION

```
2.5devel
```

### CODE REVIEW

**Comment by Akasurde** [comment]
@pdellaert @tchernomax Could you please review this ?

**Comment by Akasurde** [comment]
@pdellaert @tchernomax Could you please review this ?

**Comment by Akasurde** [comment]
ready_for_review

File: lib/ansible/modules/cloud/vmware/vmware_host_acceptance.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    - Acceptance level of this ESXi host system will be managed.
+    - If C(cluster_name) is not given, this parameter is required.
+  state:
+    description:
+    - Set or list acceptance level of the given ESXi host.
+    - 'If set to C(list), then will return current acceptance level of given host system/s.'
+    - If set to C(present), then will set given acceptance level.
+    - "If set to C(absent), then will set to default value of 'partner'."
+    choices: [ absent, list, present ]
+    required: False
```
Comment (tchernomax) [inline lib/ansible/modules/cloud/vmware/vmware_host_acceptance.py]: "state" is required or you should set a default.

File: lib/ansible/modules/cloud/vmware/vmware_host_acceptance.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+  acceptance_level:
+    description:
+    - Name of acceptance level.
+    - If set to C(partner), then accept only partner and VMware signed and certified VIBs.
+    - If set to C(vmware_certified), then accept only VIBs that are signed and certified by VMware.
+    - If set to C(vmware_accepted), then accept VIBs that have been accepted by VMware.
+    - If set to C(community), then accept all VIBs, even those that are not signed.
+    choices: [ community, partner, vmware_accepted, vmware_certified ]
```
Comment (tchernomax) [inline lib/ansible/modules/cloud/vmware/vmware_host_acceptance.py:55]: acceptance_level should be mandatory if state = present.

File: lib/ansible/modules/cloud/vmware/vmware_host_acceptance.py (hunk 1 above)
Comment (tchernomax) [reply lib/ansible/modules/cloud/vmware/vmware_host_acceptance.py]: is "absent" necessary is it's the same as "state = present" + "acceptance_level = partner" ?

File: lib/ansible/modules/cloud/vmware/vmware_host_acceptance.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+                self.hosts = [host for host in cluster_obj.host]
+            else:
+                module.fail_json(changed=False, msg="Cluster '%s' not found" % cluster_name)
+        elif esxi_host_name:
+            esxi_host_obj = self.find_hostsystem_by_name(host_name=esxi_host_name)
+            if esxi_host_obj:
+                self.hosts = [esxi_host_obj]
+            else:
+                module.fail_json(changed=False, msg="ESXi '%s' not found" % esxi_host_name)
+        self.desired_state = self.params.get('state', 'list')
```
Comment (tchernomax) [inline lib/ansible/modules/cloud/vmware/vmware_host_acceptance.py]: you should set the default state to "list" at argument_spec, not here.

File: lib/ansible/modules/cloud/vmware/vmware_host_acceptance.py [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+                host_image_config_mgr = host.configManager.imageConfigManager
+                if host_image_config_mgr:
+                    try:
+                        host_image_config_mgr.UpdateHostImageAcceptanceLevel(newAcceptanceLevel='partner')
+                        host_changed = True
+                        self.hosts_facts[host.name]['level'] = host_image_config_mgr.HostImageConfigGetAcceptance()
+                    except vim.fault.HostConfigFault as e:
+                        self.hosts_facts[host.name]['error'] = to_native(e.msg)
+            change.append(host_changed)
+        self.module.exit_json(changed=any(change), facts=self.hosts_facts)
```
Comment (tchernomax) [inline lib/ansible/modules/cloud/vmware/vmware_host_acceptance.py]: remove reset_acceptance_level ; modify and re-use set_acceptance_level instead (reset_acceptance_level and set_acceptance_level have just 1 minor difference).

**Review by tchernomax — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/vmware/vmware_host_acceptance.py (hunk 1 above)
Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_host_acceptance.py]: OK. I will remove absent.

File: lib/ansible/modules/cloud/vmware/vmware_host_acceptance.py (hunk 2 above)
Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_host_acceptance.py:55]: Done.

File: lib/ansible/modules/cloud/vmware/vmware_host_acceptance.py (hunk 3 above)
Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_host_acceptance.py]: Done.

File: lib/ansible/modules/cloud/vmware/vmware_host_acceptance.py (hunk 4 above)
Comment (Akasurde) [reply lib/ansible/modules/cloud/vmware/vmware_host_acceptance.py]: Done.

**Comment by Akasurde** [comment]
ready_for_review

**Review by tchernomax — APPROVED** [review]
LGTM

**Pull request merged**